src/modules/generation/service.py

```diff
@@ -0,0 +1,174 @@
+                )
+                content += part.choices[0].delta.content or ""
+                generation_dict["content"] = content
+
+                await cache.json().set(
+                    key,
+                    "$",
+                    generation_dict,
+                )
+                logger.info(
+                    f"service=generate message='saved response object to cache during stream handling' extra='generation_id={generationId}'"
+                )
+
+            generation_dict["status"] = GenerationStatus.COMPLETE.value
+            generation_dict["content"] = content
+
+            await cache.json().set(
+                key,
+                "$",
+                generation_dict,
+            )
+            logger.info(
+                f"service=generate message='saved response object to cache after stream handling' extra='generation_id={generationId}'"
+            )
+
+        except Exception as e:
+            generation_dict["status"] = GenerationStatus.COMPLETE.value
+            generation_dict["content"] = None
+            generation_dict["error"] = e
+
+            await cache.json().set(
+                key,
+                "$",
+                generation_dict,
+            )
+            logger.error(
+                f"service=generate, msg{e} extra='generation_id={generationId}'"
+            )
+
+        finally:
+            await cache.expireat(
+                key,
+                self.__get_session_expiry(settings.DELETE_GENERATION_AFTER_DAYS),
+            )
+
+            logger.info(
+                f"service=generate message='set cache expiry for generation request' extra='generation_id={generationId}'"
+            )
+
+    async def generation_status(self, generationId: str):
+        try:
+            cache = await get_cache()
+            data = await cache.json().get(
+                f"{settings.GENERATION_JOB_PREFFIX_KEY}:{generationId}"
+            )
+
+            if data["status"] == GenerationStatus.INCOMPLETE.value:
+                data["content"] = None
+            del data["prompt"]
+
+            return data
+        except Exception as e:
+            response = {
+                "status": GenerationStatus.INCOMPLETE.value,
+                "content": None,
+                "error": None,
+            }
+            logger.error(
+                f"service=generation_status, msg='{e}' extra='generation_id={generationId}'"
+            )
+            response["status"] = GenerationStatus.COMPLETE.value
+            response["error"] = e
+            return response
+
+    def __get_session_expiry(self, days: int) -> int:
+        current_datetime = datetime.now()
+        if days < 1:
+            hours = round(days * 24, 1)
+            days = 0
+        else:
+            hours = 0
+        expiry_datetime = current_datetime + timedelta(days=days, hours=hours)
```

Comment:
I'm not sure we need this if/else. If `days < 1` then `days = 0` and `round(days * 24, 1)` would always give us the same result. So I think it's safe enough to simply call L161  without the hours argument

That said, if we want to support setting expiry periods that aren't up to one day, then we should change the variable from `days` to `hours`, and update the logic accordingly

I think we should also rename `__get_session_expiry` to something like `__get_generation_expiry` or something else since we're not really dealing with sessions here